vGrid

MarkStega opened the pull request.

MarkStega committed:
Intermediate

MarkStega committed:
Measure text width resolution (30 seconds to 0.2 seconds)

MarkStega committed:
Intermediate

MarkStega committed:
Next three demo pages cleaned

MarkStega committed:
More demo pages

MarkStega committed:
Intermediate

MarkStega committed:
Merge branch 'main' into vGrid

MarkStega committed:
2021-01-18 EoD

MarkStega committed:
2021-01-19 EoD

MarkStega committed:
2021-01-20 EoD

MarkStega committed:
Complete #308

MarkStega committed:
Intermediate

MarkStega committed:
Grid update removing data manipulation

MarkStega committed:
intermediate

MarkStega committed:
2021-01-22 EoD

MarkStega committed:
2020-01-25 Dictionary -> List<KeyValuePair>

MarkStega committed:
2021-01-25 Added Header to MBDialog - Will remove if there is a reliable method to achieve this using CSS

MarkStega committed:
2021-01-27 EoD

MarkStega committed:
Update List<ColumnConfiguration> => IEnumerable<ColumnConfiguration>

MarkStega committed:
2021-01-28 EoD

MarkStega committed:
2021-01-29 EoD

MarkStega committed:
2021-01-31 EoD

MarkStega committed:
2021-02-01 Parameter name cleanup in MBGrid_DataHelper

stefanloerwald committed:
Snackbar (#24)

* created copy of Toast named Snackbar

* initial actual snackbar visuals

* replace API of toast with draft API of snackbar

* cleanup: MBToastAnchor is obsolete

* cleanup configuration

* fix documentation for MBAnchor

* https://material.io/components/snackbars/web#javascript-api mandates default timeout of 5000ms

* timeout related fixes

* fix types for timeout

* dismiss

* the queue of snackbars works somewhat different to that of toasts, because the snackbars timeout/close via MDC

* action is now usable; improved demo

* support MarkupString and improved demo

* correct position in sidenav

* method & property documentation

Co-authored-by: Stefan Lörwald <stefan.loerwald@gmail.com>

stefanloerwald committed:
new feature: fine control over which dates are enabled/disabled (#25)

* new feature: fine control over which dates are enabled/disabled

* fixes FirstDayOfWeek to obey CurrentCulture

* bug fix for Sunday being the first day of the week

Co-authored-by: Stefan Lörwald <stefan.loerwald@gmail.com>

stefanloerwald commented on Material.Blazor.Website/Pages/Anchor.razor:
```diff
@@ -23,9 +23,11 @@
-                    must be placed in your app.razor or the layout page used for all application components
-                    that use toasts and tooltips.</p>
+                    <p>
+                        The Anchor is a non-visual component. If you use Toast and/or Tooltip components this
```
Snackbar should be included here too.

stefanloerwald commented on Material.Blazor.Website/Pages/ChipsSelectMulti.razor:
```diff
@@ -59,10 +73,10 @@
         new MBIconBearingSelectElement<string> { SelectedValue = "brit-short", Label = "British Shorthair" },
         new MBIconBearingSelectElement<string> { SelectedValue = "russ-blue", Label = "Russian Blue", Icon="admin_panel_settings" },
         new MBIconBearingSelectElement<string> { SelectedValue = "ice-invis", Label = "Icelandic Invisible", Icon="card_membership" }
-    };
```
Razor formatting issue?

stefanloerwald commented on Material.Blazor.Website/Pages/ChipsSelectSingle.razor:
```diff
@@ -60,7 +66,7 @@
         new MBIconBearingSelectElement<string> { SelectedValue = "brit-short", Label = "British Shorthair" },
         new MBIconBearingSelectElement<string> { SelectedValue = "russ-blue", Label = "Russian Blue", Icon="admin_panel_settings" },
         new MBIconBearingSelectElement<string> { SelectedValue = "ice-invis", Label = "Icelandic Invisible", Icon="card_membership" }
-    };
```
razor formatting issue?

stefanloerwald committed:
new feature: allow custom values for auto-complete (#26)

* new feature: allow custom values for auto-complete

* demonstrate how the custom style can be applied

Co-authored-by: Stefan Lörwald <stefan.loerwald@gmail.com>

MarkStega committed:
2021-03-03 - Format AutocompleteTextField.razor

MarkStega committed:
2021-02-03 Update component razor files to follow the style of alphabetized single line

MarkStega committed:
Repaired autoformatted/autocorrected MBPagedDataList.razor that Visual Studio mangled.

stefanloerwald committed:
Select fix (#27)

* bug fix: items should respect the disabled setting

* shuffle blocks to make the demo page a bit neater

* formatting: one parameter per line

Co-authored-by: Stefan Lörwald <stefan.loerwald@gmail.com>

MarkStega committed:
2020-02-04 Alphabetized parameters in Select.razor

MarkStega committed:
2021-02-04 - Really alphabetize Select.razor

stefanloerwald committed:
all interfaces should be public to be mockable (#28)

Co-authored-by: Stefan Lörwald <stefan.loerwald@gmail.com>

stefanloerwald committed:
Formatting (#29)

* using dotnet format with .editorconfig provided by Simon

* adjusted the config to revert to the apparent convention of newline at the end of file

Co-authored-by: Stefan Lörwald <stefan.loerwald@gmail.com>

simonziegler committed:
Review Updates (#30)

SZ and MS to walk through review, in particular for docs. Once review is complete SL to upgrade Snackbar configuration and settings.

Co-authored-by: MarkStega <mark2002@stega.us>

MarkStega committed:
2020-02-08 Edit of edits complete

stefanloerwald committed:
Snackbar (#31)

* fix unrelated error in markup

* made all settings of a snackbar configurable via the service

* use default config in demo app

* fix documentation (add snackbar where appropriate, fix copy paste errors)

Co-authored-by: Stefan Lörwald <stefan.loerwald@gmail.com>

stefanloerwald committed:
Snackbar (#32)

* fix unrelated error in markup

* made all settings of a snackbar configurable via the service

* use default config in demo app

* fix documentation (add snackbar where appropriate, fix copy paste errors)

* changed wording in demo

Co-authored-by: Stefan Lörwald <stefan.loerwald@gmail.com>

simonziegler reviewed (approved):
Carefully checked all updates.
